src/components/layout/nav/ToolbarMenu.tsx

```diff
@@ -65,60 +65,32 @@ export const useActiveNaveItem = () => {
 };
 
 const ToolbarMenu: React.FC = () => {
-  const activeItem = useActiveNaveItem();
+  const activeItem = useActiveNavItem();
   const isMobile = useIsMobile();
 
-  const navItemSx = useCallback(
-    (selected: boolean): SxProps<Theme> => {
-      return isMobile
-        ? {
-            py: 1,
-            px: 3,
-            textDecoration: 'none',
-            textOverflow: 'ellipsis',
-            overflowX: 'hidden',
-            whiteSpace: 'nowrap',
-            display: 'block',
-            color: selected ? 'primary.main' : 'text.primary',
-            fontWeight: selected ? 600 : 400,
-            '&.Mui-focusVisible': {
-              outlineOffset: '-2px',
-            },
-            backgroundColor: selected
-              ? (theme: Theme) => lighten(theme.palette.primary.light, 0.9)
-              : undefined,
-            border: '2px solid transparent',
-          }
-        : {
-            fontWeight: 600,
-            fontSize: 14,
-            px: { xs: 0.5, lg: 2 },
-            color: 'text.primary',
-            backgroundColor: selected
-              ? (theme: Theme) => alpha(theme.palette.primary.light, 0.12)
-              : undefined,
-          };
-    },
-    [isMobile]
-  );
-
   return TOOLBAR_MENU_ITEMS.map((item) => {
     let navItem = isMobile ? (
-      <MenuItem
-        component={RouterLink}
-        to={item.path}
-        sx={navItemSx(activeItem === item.activeItemPathIncludes)}
-        data-testid={item.id}
+      <MobileMenuItem
         key={item.id}
-      >
-        {item.title}
-      </MenuItem>
+        title={item.title as string}
+        selected={activeItem === item.activeItemPathIncludes}
+        path={item.path}
+      />
     ) : (
       <ButtonLink
         variant='text'
         to={item.path}
         data-testid={item.id}
-        sx={navItemSx(activeItem === item.activeItemPathIncludes)}
+        sx={{
+          fontWeight: 600,
+          fontSize: 14,
```

Comment:
When we have custom typography, maybe we should start calling out with a FIXME comment so we can clean it up later. Ideally we shouldn't doing this